From eb8c76d451f45300a15b1cb880915fca51dacb2a Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:12:50 -0500
Subject: [ethernet] be2net: replace strcpy with strlcpy

Message-id: <1416413587-12091-3-git-send-email-ivecera@redhat.com>
Patchwork-id: 100326
O-Subject: [RHEL7.1 PATCH 02/19] be2net: replace strcpy with strlcpy
Bugzilla: 1165755
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: #1165755

Upstream commit(s):
commit 242eb47010840056c121c1141bddf3b279cb47bc
Author: Vasundhara Volam <vasundhara.volam@emulex.com>
Date:   Fri Sep 12 17:39:15 2014 +0530

    be2net: replace strcpy with strlcpy

    Replace strcpy with strlcpy, as it avoids a possible buffer overflow.

    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_cmds.c b/drivers/net/ethernet/emulex/benet/be_cmds.c
index 1fcd69e..5d8016c 100644
--- a/drivers/net/ethernet/emulex/benet/be_cmds.c
+++ b/drivers/net/ethernet/emulex/benet/be_cmds.c
@@ -1770,8 +1770,10 @@ int be_cmd_get_fw_ver(struct be_adapter *adapter)
  if (!status) {
   struct be_cmd_resp_get_fw_version *resp = embedded_payload(wrb);
 
-  strcpy(adapter->fw_ver, resp->firmware_version_string);
-  strcpy(adapter->fw_on_flash, resp->fw_on_flash_version_string);
+  strlcpy(adapter->fw_ver, resp->firmware_version_string,
+   sizeof(adapter->fw_ver));
+  strlcpy(adapter->fw_on_flash, resp->fw_on_flash_version_string,
+   sizeof(adapter->fw_on_flash));
  }
 err:
  spin_unlock_bh(&adapter->mcc_lock);
@@ -2214,7 +2216,7 @@ int lancer_cmd_write_object(struct be_adapter *adapter, struct be_dma_mem *cmd,
 
  be_dws_cpu_to_le(ctxt, sizeof(req->context));
  req->write_offset = cpu_to_le32(data_offset);
- strcpy(req->object_name, obj_name);
+ strlcpy(req->object_name, obj_name, sizeof(req->object_name));
  req->descriptor_count = cpu_to_le32(1);
  req->buf_len = cpu_to_le32(data_size);
  req->addr_low = cpu_to_le32((cmd->dma +
@@ -2267,7 +2269,7 @@ int lancer_cmd_delete_object(struct be_adapter *adapter, const char *obj_name)
           OPCODE_COMMON_DELETE_OBJECT,
           sizeof(*req), wrb, NULL);
 
- strcpy(req->object_name, obj_name);
+ strlcpy(req->object_name, obj_name, sizeof(req->object_name));
 
  status = be_mcc_notify_wait(adapter);
 err:
-- 
1.7.1